submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestReleaser.kt

```diff
@@ -111,6 +119,54 @@ class SubmissionRequestReleaser(
         return releasedFile
     }
 
+    private suspend fun suppressRequest(
+        accNo: String,
+        request: SubmissionRequest,
+    ) {
+        val sub = request.submission
+        logger.info { "$accNo ${sub.owner} Started suppressing submission files over ${sub.storageMode}" }
+        suppressSubmissionFiles(sub)
+        logger.info { "$accNo ${sub.owner} Finished suppressing submission files over ${sub.storageMode}" }
+    }
+
+    private suspend fun suppressSubmissionFiles(sub: ExtSubmission) {
+        suspend fun suppressFile(reqFile: SubmissionRequestFile) {
+            when (val file = reqFile.file) {
+                is NfsFile -> {
+                    val released = reqFile.copy(file = suppress(sub, reqFile.index, file), status = SUPPRESSED)
```

Comment:
I think is incorrect to use this index as this is the index in the submission not the index in the total files we need to supress

Reply:
For a public submission that goes private, the amount of files that need to be un-released are the total files of the submission aren't they?

Reply:
yes that is correct for that case but is not correct for others for example if a submission if modifieid and re submitted (to make files private).